extensions/amp-accordion/amp-accordion.md

```diff
@@ -267,6 +267,18 @@ it collapses all sections of the accordion. To specify a section, add the
 </button>
 ```
 
+### Actions in Bento Mode
+
+In Bento mode, the above actions can be called via `getApi()`. For example:
+
+```html
+<script>
+  var api = await document.querySelector('#myAccordion').getApi();
+  api.toggle();
+  api.toggle('section1');
+</script>
```

Comment:
I'm a little confused about this section - we're getting the API, but I don't understand what is executing the actions. Do we have an example of what that might look like?

Reply:
So the idea is that the actions can be executed programatically, (i.e. javascript code can directly access the elements and run the actions.).

So in this section the first line `var api = ....getApi()`, gets an API object from our accordion bento component.  This can be called on all bento components.  The returned API object will be unique to the type of component.  In the case of accordion, there are three actions we have defined in the API: `toggle`, `expand`, and `collapse`.

An example is shown calling `toggle` with no arguments and then `toggle` with `section1` as an argument.  The behavior of what these would do could be defined in the accordion documentation since its unique to accordion (not all components will have a `toggle` action.).  Lightbox might have an `open` and `close` action to open and close the lightbox for example.

In terms of what's executing the action, it's a javascript code that's directly included in the bento page (in this example).  Although theoretically any code that has access to the bento components can execute the actions, the primary example we have so far is javascript code on the page (inside the `<script>` tag, javascript code is run)

Another way to try this out is to open the browser console and write the above code directly in it.  (cmd + option + j on mac, chrome)

Reply:
Thanks for the explanation! But what I would really love to see is a working code example (html & js)  of a non-AMP page element triggering one of these events. Such as clicking a button outside the amp-accordion that would invoke the `toggle`. The current code snippet looks like it gives the page access to the API, but I don't see how it uses what it makes available.

Reply:
Thanks for the clarification.  Here's an example that we can use that is a bit more comprehensive.  I've included some comments in here that may be useful to the user, but we can remove in these for our example or explain it elsewhere.

```
<amp-accordion id="myAccordion" animate='true'>
  <section id="section1">
    <h2>Section 1</h2>
    <div>Puppies are cute.</div>
  </section>
  <section id="section2">
    <h2>Section 2</h2>
    <div>Kittens are furry.</div>
  </section>
  <section id="section3">
    <h2>Section 3</h2>
    <div>Elephants have great memory.</div>
  </section>
</amp-accordion>

<div class="buttons" style="margin-top: 8px;">
  <button id='button1'>Toggle All Sections</button>
  <button id='button2'>Expand Section 2</button>
  <button id='button3'>Collapse Section 3</button>
</div>

<script>
  (async () => {
    const accordion = document.querySelector('#myAccordion');
    await customElements.whenDefined('amp-accordion');
    const api = await accordion.getApi();
    
    // programatically toggle (expands) all sections
    api.toggle();

    // set up button actions
    document.querySelector('#button1').onclick = () => api.toggle();
    document.querySelector('#button2').onclick = () => api.expand('section2');
    document.querySelector('#button3').onclick = () => api.collapse('section3');
  })();
</script>
```

Reply:
Thank you! This is a perfect example - I think it should be included in the documentation. Since it isn't valid AMP we might need to keep it as just a snippet and not an inline executable example cc/ @sebastianbenz

Reply:
Wonderful example! Let's do as an executable sample, but using the standalone preview.